x-pack/plugins/maps/public/connected_components/add_layer_panel/flyout_footer/index.ts

```diff
@@ -7,22 +7,24 @@
 import { AnyAction, Dispatch } from 'redux';
 import { connect } from 'react-redux';
 import { FlyoutFooter } from './view';
-import { getSelectedLayer } from '../../../selectors/map_selectors';
-import { clearTransientLayerStateAndCloseFlyout } from '../../../actions';
+import { hasPreviewLayers, isLoadingPreviewLayers } from '../../../selectors/map_selectors';
+import { removePreviewLayers, updateFlyout } from '../../../actions';
 import { MapStoreState } from '../../../reducers/store';
+import { FLYOUT_STATE } from '../../../reducers/ui';
 
 function mapStateToProps(state: MapStoreState) {
-  const selectedLayer = getSelectedLayer(state);
-  const hasLayerSelected = !!selectedLayer;
   return {
-    hasLayerSelected,
-    isLoading: hasLayerSelected && selectedLayer!.isLayerLoading(),
+    hasPreviewLayers: hasPreviewLayers(state),
+    isLoading: isLoadingPreviewLayers(state),
   };
 }
 
 function mapDispatchToProps(dispatch: Dispatch<AnyAction>) {
   return {
-    closeFlyout: () => dispatch<any>(clearTransientLayerStateAndCloseFlyout()),
+    closeFlyout: () => {
+      dispatch(updateFlyout(FLYOUT_STATE.NONE));
+      dispatch<any>(removePreviewLayers());
```

Comment:
There are a few other places where we're closing the flyout but not removing preview layers. We might want to create an action called something like `closeFlyoutAndClearTempLayers` that calls `updateFlyout(FLYOUT_STATE.NONE)` and also clears all preview and selected layers so we don't miss any cases.

Reply:
Where are those places? I suspect that in those places, it is not possible to have preview layers. In the last couple of minors, we have updated the UI to avoid opening flyout when its already opened to avoid having to do weird clean-ups like this.

Reply:
> I suspect that in those places, it is not possible to have preview layers

That might be obvious now but may not be as obvious for devs that come after. It's difficult to know for sure which flyout close actions should coincide with what clean up, whereas I find it very easy to reason about:

An action that cancels or completes layer adding can always:
- clean up any temp layers since temp layers are an add layer construct
- close flyout

All that said, this doesn't need to hold up this PR. This might be more of an offline architectural discussion.